homeassistant/components/elkm1/binary_sensor.py

```diff
@@ -0,0 +1,66 @@
+"""Support for control of ElkM1 binary sensors."""
+from __future__ import annotations
+
+from typing import Any
+
+from elkm1_lib.const import ZoneLogicalStatus, ZoneType
+from elkm1_lib.elements import Element
+from elkm1_lib.elk import Elk
+from elkm1_lib.zones import Zone
+
+from homeassistant.components.binary_sensor import BinarySensorEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import ElkAttachedEntity, ElkEntity
+from .const import DOMAIN
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Create the Elk-M1 sensor platform."""
+
+    elk_data = hass.data[DOMAIN][config_entry.entry_id]
+    auto_configure = elk_data["auto_configure"]
+    elk = elk_data["elk"]
+
+    entities: list[ElkEntity] = []
+    for element in elk.zones:
+        # Don't create binary sensors for zones that are analog
+        if element.definition in {ZoneType.TEMPERATURE, ZoneType.ANALOG_ZONE}:
+            continue
+
+        if auto_configure:
+            if not element.configured:
+                continue
+        elif not elk_data["config"]["zone"]["included"][element.index]:
+            continue
+
+        entities.append(ElkBinarySensor(element, elk, elk_data))
+
+    async_add_entities(entities, True)
```

Comment:
```suggestion
    async_add_entities(entities)
```

Do we need `update_before_add=True` ?

Reply:
Good question. The rest of the integration used `update_before_add` and has been that way since day 1. What does the flag do?

Reply:
I have limited understanding of what the update flag does, but it appear to be when the name of the entity changes, which does not seem applicable. Feels like I can remove it, but you know best here.

Reply:
Its going to call `async_device_update` before adding the entity to Home Assistant 

https://github.com/home-assistant/core/blob/ce35324e7391f0e06c415f414e7a52276ff4d1d9/homeassistant/helpers/entity_platform.py#L443

which calls `async_update` on the entity

https://github.com/home-assistant/core/blob/ce35324e7391f0e06c415f414e7a52276ff4d1d9/homeassistant/helpers/entity.py#L695

Which I'm pretty sure isn't implemented because everything comes in via callbacks

So I think we can take all of them out

Reply:
I did a quick test to verify. Taking them seems just fine and should make startup a wee bit faster